### ISSUE TITLE
macOS vulkaninfo reports 4 GPUs on a 2-GPU MacBook, unreliable surfaces

### ISSUE THREAD

**Issue by lunarpapillo** [KhronosGroup/Vulkan-Tools#316]
There's a lot of strangeness here.
```
$ cd ~/Downloads/vulkansdk-macos-1.1.130.0
$ ./Applications/vulkaninfo.app/Contents/MacOS/vulkaninfo --html
objc[1885]: Class MVKBlockObserver is implemented in both /Users/lunarg/Downloads/vulkansdk-macos-1.1.130.0/Applications/vulkaninfo.app/Contents/Frameworks/libMoltenVK.dylib (0x108e136e0) and /usr/local/lib/libMoltenVK.dylib (0x109a486e0). One of the two will be used. Which one is undefined.
$ open vulkaninfo.html
```
I think the `libMoltenVK.dylib` warning is harmless, as both versions are identical (but I include the warning here because this could feasibly cause problems). 

On one run, `vulkaninfo` reported four GPUs and eight presentable surfaces, two per GPU, the first four of which have seven surface types each, the last four of which have one: [vulkaninfo-app-1.1.130.html.txt](https://github.com/KhronosGroup/Vulkan-Tools/files/3985823/vulkaninfo-app-1.1.130.html.txt)

![vulkaninfo-app-1 1 130](https://user-images.githubusercontent.com/45772930/71213355-81f1a400-2270-11ea-9d32-09a2f9ee349e.png)

A subsequent run reported the same four GPUs, but only four presentable surfaces, each of which had *eight* surface types each:
[vulkaninfo-app-1.1.130-2.html.txt](https://github.com/KhronosGroup/Vulkan-Tools/files/3985870/vulkaninfo-app-1.1.130-2.html.txt)

![vulkaninfo-1 1 130-2](https://user-images.githubusercontent.com/45772930/71214883-823f6e80-2273-11ea-9af6-56506dd5f986.png)

So there seems to be some run-to-run variance.

More in the next response...

**Comment by lunarpapillo** [comment]
More strangeness...

Running the version shipped in `macOS/bin/vulkaninfo` seems to halve the output, showing the correct 2 GPUs, each with a single surface, but each of those surfaces with an unexpected four surface types:
```
$ ./macOS/bin/vulkaninfo --html
$ open vulkaninfo.html
```
[vulkaninfo-bin-1.1.130.html.txt](https://github.com/KhronosGroup/Vulkan-Tools/files/3985877/vulkaninfo-bin-1.1.130.html.txt)

![vulkaninfo-1 1 130-bin](https://user-images.githubusercontent.com/45772930/71215426-ef9fcf00-2274-11ea-96f5-86a18ef81b54.png)

This seems to be stable.

**Comment by lunarpapillo** [comment]
For comparison, I'm trying the system-installed libraries from SDK 1.1.130 and the `vulkaninfo` executables from SDK 1.1.121:
```
$ cd ~/Downloads/vulkansdk-macos-1.1.121.1
$ ./Applications/vulkaninfo.app/Contents/MacOS/vulkaninfo --html
objc[1907]: Class MVKBlockObserver is implemented in both /Users/lunarg/Downloads/vulkansdk-macos-1.1.121.1/Applications/vulkaninfo.app/Contents/Frameworks/libMoltenVK.dylib (0x10e26b8f0) and /usr/local/lib/libMoltenVK.dylib (0x10e8c96e0). One of the two will be used. Which one is undefined.
$ open vulkaninfo.html
```
Now, in this case, the `libMoltenVK.dylib` in the package is different from the one in the system.  In this case, I also see 4 GPUs, but each GPU only has a single surface and a single surface type:
[vulkaninfo-app-1.1.121.html.txt](https://github.com/KhronosGroup/Vulkan-Tools/files/3985916/vulkaninfo-app-1.1.121.html.txt)

![vulkaninfo-1 1 121-app](https://user-images.githubusercontent.com/45772930/71215809-00047980-2276-11ea-9acb-613ec8cf11d9.png)

Using the `macOS/bin/vulkaninfo` version (which runs using the system SDK libraries from 1.1.130) shows the correct 2 GPUs, each with one presentable surface, with only one surface type:
[vulkaninfo-bin-1.1.121.html.txt](https://github.com/KhronosGroup/Vulkan-Tools/files/3985925/vulkaninfo-bin-1.1.121.html.txt)

![vulkaninfo-1 1 121-bin](https://user-images.githubusercontent.com/45772930/71215997-6d180f00-2276-11ea-9825-8d5e67a332e6.png)

It seems to me that the `Applications/vulkaninfo.app` version (under both 1.1.121 and 1.1.130) incorrectly reports double the GPUs, as compared to the `macOS/bin/vulkaninfo` version.  It also seems that the 1.1.130 version is incorrectly reporting multiples of surface types (though it may be correct that 1.1.130 supports `VK_EXT_metal_surface` that 1.1.121 did not).

**Comment by charles-lunarg** [comment]
Closing the issue as it was resolved with PR #331.
This was due to multiplexing the surfaces with the available GPU's which shouldn't of been the case.
